From d3d73d25e0d9bc43fd2a6f4b4e58ff182e55b217 Mon Sep 17 00:00:00 2001
Message-Id: <d3d73d25e0d9bc43fd2a6f4b4e58ff182e55b217.1651196606.git.geliang.tang@suse.com>
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Tue, 17 Nov 2020 10:22:51 +0300
Subject: [PATCH] platform/x86: pmt: Fix a potential Oops on error in probe
Git-commit: d3d73d25e0d9bc43fd2a6f4b4e58ff182e55b217
Patch-mainline: v5.11-rc1
References: bsn#210
Modified-by-SEL: No

The "ns->attr_grp" pointer can be NULL so this error handling code needs
to check for that to avoid an Oops.

Fixes: e2729113ce66 ("platform/x86: Intel PMT class driver")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Reviewed-by: David E. Box <david.e.box@linux.intel.com>
Link: https://lore.kernel.org/r/20201117072251.GC1111239@mwanda
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/platform/x86/intel_pmt_class.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel_pmt_class.c b/drivers/platform/x86/intel_pmt_class.c
index aa88dc23bbde..c8939fba4509 100644
--- a/drivers/platform/x86/intel_pmt_class.c
+++ b/drivers/platform/x86/intel_pmt_class.c
@@ -225,7 +225,8 @@ static int intel_pmt_dev_register(struct intel_pmt_entry *entry,
 		return 0;
 
 fail_ioremap:
-	sysfs_remove_group(entry->kobj, ns->attr_grp);
+	if (ns->attr_grp)
+		sysfs_remove_group(entry->kobj, ns->attr_grp);
 fail_sysfs:
 	device_unregister(dev);
 fail_dev_create:
-- 
2.34.1

